[JIT] fix opinfo utils to handle tensor kwargs

davidberard98 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #76768
* #75917

Previously, the traced function would only take tensor args (and not tensor kwargs). Then the tensor kwargs would be inlined as constants into the graph during tracing, and changes to the kwarg inputs wouldn't be reflected in new calls to the traced function.

davidberard98 committed:
[JIT] fix opinfo utils to handle tensor kwargs

Previously, the traced function would only take tensor args (and not tensor kwargs). Then the tensor kwargs would be inlined as constants into the graph during tracing, and changes to the kwarg inputs wouldn't be reflected in new calls to the traced function.

[ghstack-poisoned]

davidberard98 committed:
Update on "[JIT] fix opinfo utils to handle tensor kwargs"


Previously, the traced function would only take tensor args (and not tensor kwargs). Then the tensor kwargs would be inlined as constants into the graph during tracing, and changes to the kwarg inputs wouldn't be reflected in new calls to the traced function.

[ghstack-poisoned]

mruberry commented on torch/testing/_internal/jit_metaprogramming_utils.py:
```diff
@@ -407,45 +407,78 @@ def script_fn(*args, **kwargs):
-# applied, and all tensor arguments remain.
-# used to trace functions when some arguments are not tensors
-def partial_apply_nontensors(fn, args, **kwargs):
+class SplitInputs():
```
Comment explaining this class and its intended use would be nice

mruberry reviewed (approved):
Getting rid of skips is nice

davidberard98 committed:
Update on "[JIT] fix opinfo utils to handle tensor kwargs"


Previously, the traced function would only take tensor args (and not tensor kwargs). Then the tensor kwargs would be inlined as constants into the graph during tracing, and changes to the kwarg inputs wouldn't be reflected in new calls to the traced function.

[ghstack-poisoned]

davidberard98 committed:
Update on "[JIT] fix opinfo utils to handle tensor kwargs"


Previously, the traced function would only take tensor args (and not tensor kwargs). Then the tensor kwargs would be inlined as constants into the graph during tracing, and changes to the kwarg inputs wouldn't be reflected in new calls to the traced function.

[ghstack-poisoned]

davidberard98 committed:
Update on "[JIT] fix opinfo utils to handle tensor kwargs"


Previously, the traced function would only take tensor args (and not tensor kwargs). Then the tensor kwargs would be inlined as constants into the graph during tracing, and changes to the kwarg inputs wouldn't be reflected in new calls to the traced function.

[ghstack-poisoned]

davidberard98 commented:
@pytorchbot merge this please
